pkg/install/stack/common/validation.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2020 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+)
+
+func ValidateRandomSecret(value string, path string) error {
+	if value == "" {
+		secret, err := randomString()
+		if err == nil {
```

Comment:
In which cases this can return an error? This function just generates a random string, it doesn't depend on `value` or `path`, so I don't see in which cases it can fail.

Reply:
Erm it's literally defined right below that:

```
func randomString() (string, error) {
	c := 32
	b := make([]byte, c)

	_, err := rand.Read(b) // <-- this can fail
	if err != nil {
		return "", err
	}

	return base64.RawURLEncoding.EncodeToString(b), nil
}
```

Reply:
I've seen this, but what confuses me is why we have `err == nil` instead of `err != nil` on L28. Considering https://github.com/kubermatic/kubermatic/pull/6435#discussion_r571913658, we want this code to generate the random string if there's no string provided, so I think that we should return an error only if that fails.

Reply:
Oh. Ooooooohohohooo, I see. :-D

Reply:
No I don't. The current code is correct, the `err == nil` is what I wanted to express there. :)

Reply:
We've discussed it offline and it makes sense now.